New authentication does not work on MacOS

**Describe the bug**
When I try to join the created server, the world downloader cannot get login details from the minecraft process (possibly because I'm on macOS)

**Steps to reproduce**
Simply try to join a world

**Log output**
```
Client probably disconnected. Waiting for new connection...
Something went wrong while trying to authenticate your Minecraft account.

Authentication details are retrieved from the Minecraft process's launch arguments. The game must be running for this to work.
Alternatively, use options --username and --token to specify the details manually.
See https://github.com/mircokroon/minecraft-world-downloader/wiki/Authentication for more details.

javax.security.sasl.AuthenticationException: Cannot get valid authentication details. [Caused by javax.security.sasl.AuthenticationException: Cannot find authentication details.]
	at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:40)
	at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$6(EncryptionManager.java:225)
	at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:113)
	at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:225)
	at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$5(EncryptionManager.java:213)
	at proxy.EncryptionManager.attempt(EncryptionManager.java:100)
	at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:202)
	at packets.handler.ServerBoundLoginPacketHandler.lambda$new$1(ServerBoundLoginPacketHandler.java:30)
	at packets.handler.PacketHandler.handle(PacketHandler.java:57)
	at packets.DataReader.readPackets(DataReader.java:163)
	at packets.DataReader.pushData(DataReader.java:118)
	at proxy.ProxyServer.lambda$run$4(ProxyServer.java:86)
	at util.ExceptionHandling.attempt(ExceptionHandling.java:12)
	at proxy.ProxyServer.lambda$run$6(ProxyServer.java:83)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: javax.security.sasl.AuthenticationException: Cannot find authentication details.
	at proxy.auth.AuthDetailsManager.getAuthDetails(AuthDetailsManager.java:47)
	at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:37)
	... 14 more
Client probably disconnected. Waiting for new connection..

```

**Version**
 - Game: Minecraft 1.17.1
 - Downloader: latest

--- a/src/main/java/proxy/auth/AuthDetailsFromProcess.java
+++ b/src/main/java/proxy/auth/AuthDetailsFromProcess.java
@@ -71,9 +71,6 @@ public class AuthDetailsFromProcess {
                     .map(this::getLaunchParametersWindows)
                     .filter(line -> line.contains("--accessToken"))
                     .collect(Collectors.toList());
-        } else if (SystemUtils.IS_OS_MAC_OSX) {
-            return findCandidateProcessMacOS().stream()
-            .map(a -> "" + a).collect(Collectors.toList());
         } else {
             return findCandidateProcessUnix().stream()
                     .map(a -> "" + a).collect(Collectors.toList());
@@ -103,33 +100,17 @@ public class AuthDetailsFromProcess {
     }
 
     /**
-     * For unix (non-macOS) systems we can use one command to get all java processes and their arguments. The first line will be a
+     * For unix systems we can use one command to get all java processes and their arguments. The first line will be a
      * header that we can read out to get the required index for the arguments.
      */
     private List<String> findCandidateProcessUnix() throws IOException {
-        Process p = Runtime.getRuntime().exec("ps -fC java");
-
-        BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()));
-
-        List<String> res = new ArrayList<>();
-
-        String line;
-        while ((line = input.readLine()) != null) {
-            if (line.contains("java") && line.contains("--accessToken")) {
-                res.add(line);
-            }
+        Process p;
+        if (SystemUtils.IS_OS_MAC_OSX) {
+            p = Runtime.getRuntime().exec("ps ax");
         }
-        input.close();
-
-        return res;
-    }
-
-    /**
-     * macOS systems have a different version of ps installed which requires different arguments,
-     * but we can just get the entire process list and filter through it.
-     */
-    private List<String> findCandidateProcessMacOS() throws IOException {
-        Process p = Runtime.getRuntime().exec("ps ax");
+        else {
+            p = Runtime.getRuntime().exec("ps -fC java");
+        }
 
         BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()));
 

--- a/src/main/java/proxy/auth/AuthDetailsFromProcess.java
+++ b/src/main/java/proxy/auth/AuthDetailsFromProcess.java
@@ -71,6 +71,9 @@ public class AuthDetailsFromProcess {
                     .map(this::getLaunchParametersWindows)
                     .filter(line -> line.contains("--accessToken"))
                     .collect(Collectors.toList());
+        } else if (SystemUtils.IS_OS_MAC_OSX) {
+            return findCandidateProcessMacOS().stream()
+            .map(a -> "" + a).collect(Collectors.toList());
         } else {
             return findCandidateProcessUnix().stream()
                     .map(a -> "" + a).collect(Collectors.toList());
@@ -100,11 +103,33 @@ public class AuthDetailsFromProcess {
     }
 
     /**
-     * For unix systems we can use one command to get all java processes and their arguments. The first line will be a
+     * For unix (non-macOS) systems we can use one command to get all java processes and their arguments. The first line will be a
      * header that we can read out to get the required index for the arguments.
      */
     private List<String> findCandidateProcessUnix() throws IOException {
         Process p = Runtime.getRuntime().exec("ps -fC java");
+
+        BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()));
+
+        List<String> res = new ArrayList<>();
+
+        String line;
+        while ((line = input.readLine()) != null) {
+            if (line.contains("java") && line.contains("--accessToken")) {
+                res.add(line);
+            }
+        }
+        input.close();
+
+        return res;
+    }
+
+    /**
+     * macOS systems have a different version of ps installed which requires different arguments,
+     * but we can just get the entire process list and filter through it.
+     */
+    private List<String> findCandidateProcessMacOS() throws IOException {
+        Process p = Runtime.getRuntime().exec("ps ax");
 
         BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()));
 

